autotuning/tcl/apollo10_eyescan.tcl

```diff
@@ -0,0 +1,195 @@
+set mgt_tx_list [eval get_hw_sio_txs]
+set mgt_rx_list [eval get_hw_sio_rxs]
+set mgt_len [llength $mgt_tx_list]
+set mgt_link_list [eval get_hw_sio_links]
+##remove_hw_sio_link [get_hw_sio_links]
+##set Txs {20,59}
+##set Rxs {43,4}
+
+set date 06-08-22
+#Example date 01-19-22
+#Also, be sure to create corresponding directory to save the scans to (e.g. /mnt/scratch/ad683/Cornell_CM_Production_Scripts/scans/CM203/01-19-22)
+
+## Links between FPGA (Tx are the ones from FPGA1, Rxs are from FPGA2)
+set Txs {}
+set Rxs {}
+
+for {set t 20} {$t<60} {incr t} {
+    set tstring "X1Y$t/"
+    lappend Txs $tstring
+}
+
+for {set r 43} {$r>3} {incr r -1} {
+    set rstring "X1Y$r/"
+    lappend Rxs $rstring                                                                                                                                                          
+}
+
+for {set t 4} {$t<16} {incr t} {
+    set tstring "X1Y$t/"
+    lappend Txs $tstring
+}
+
+for {set r 59} {$r>47} {incr r -1} {
+    set rstring "X1Y$r/"
+    lappend Rxs $rstring
+}
+
+## FFs FPGA1 - FPGA2
+for {set t 40} {$t<44} {incr t} {
+    set tstring "X0Y$t/"
+    lappend Txs $tstring
+}
+
+for {set r 32} {$r<36} {incr r} {
+    set rstring "X0Y$r/"
+    lappend Rxs $rstring
+}
+
+set i 0
+foreach Tx $Txs Rx $Rxs {
+    puts "MGT $i"
+    puts [lsearch -all -inline $mgt_link_list "*2542/0_1*$Tx*->*2542/1_1*$Rx*"]
+    set xil_newScan [create_hw_sio_scan -description "Scan $i" 2d_full_eye  [lindex [get_hw_sio_links [lsearch -all -inline $mgt_link_list "*2542/0_1*$Tx*->*2542/1_1*$Rx*"]] 0 ]]
+    set_property HORIZONTAL_INCREMENT {1} [get_hw_sio_scans $xil_newScan]
+    set_property VERTICAL_INCREMENT {1} [get_hw_sio_scans $xil_newScan]
+    run_hw_sio_scan [get_hw_sio_scans $xil_newScan]
+    incr i 1
+    wait_on_hw_sio_scan [get_hw_sio_scans $xil_newScan]
+    set trimTx [string trim $Tx "/"]
+    set trimRx [string trim $Rx "/"]
+    #write_hw_sio_scan -force "/mnt/scratch/ad683/Cornell_CM_Production_Scripts/autotuning/eyescan_results/eyescan_${trimTx}_to_${trimRx}" [get_hw_sio_scans $xil_newScan]
+    write_hw_sio_scan -force "/mnt/scratch/ad683/Cornell_CM_Production_Scripts/scans/CM203/${date}/eyescan_${trimTx}(xcvu13p_0)_to_${trimRx}(xcvu13p_1)" [get_hw_sio_scans $xil_newScan]
```

Comment:
Directory is now configurable at the beginning of the file.  However, I don't recommend changing the general structure of it because the rest of the files require files to be saved in a particular directory structure in order to function.